### PR TITLE
Handle SEO rule in metadata validation

### PR DESCRIPTION
**Pull request by fkpwolf**
[AB#397298](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/397298)

### CODE REVIEW

File: src/docfx/validation/CustomRuleProvider.cs [hunk 1]
```diff
@@ -48,6 +52,17 @@ public bool IsEnable(FilePath filePath, CustomRule customRule, string? moniker =
                 return false;
             }
 
+            if (customRule.Tags != null && customRule.Tags.Contains("SEO"))
+            {
+                var metadata = _metadataProvider.GetMetadata(ErrorBuilder.Null, filePath).RawJObject;
+                var noindex = metadata.TryGetValue<JValue>("robot", out var robotValue) && robotValue.Value is string robot &&
```
Comment (yufeih) [inline src/docfx/validation/CustomRuleProvider.cs:58]: prefer `UserMetadata.Robots`

File: src/docfx/validation/CustomRuleProvider.cs [hunk 2]
```diff
@@ -48,6 +52,17 @@ public bool IsEnable(FilePath filePath, CustomRule customRule, string? moniker =
...
+                    robot.Contains("noindex", StringComparison.OrdinalIgnoreCase);
+                if (noindex)
```
Comment (yufeih) [inline src/docfx/validation/CustomRuleProvider.cs:60]: extract a `UserMetadata.NoIndex()` method?
image

File: src/docfx/validation/CustomRuleProvider.cs (hunk 1 above)
Comment (fkpwolf) [reply src/docfx/validation/CustomRuleProvider.cs:58]: `UserMetadata.Robots` is null here. And I find some code like this:
```
_metadataValidator.ValidateMetadata(errors, userMetadata.RawJObject, file);
```

File: docs/specs/validation/metadata-validation.yml [hunk 3]
```diff
@@ -452,3 +452,89 @@ outputs:
...
+            "tags": [ "SEO"]
+          }
+        }
+      }
+    }
+  a.md: |
+    ---
+    title: title1
+    description: description1
+    robot: noindex
```
Comment (yufeih) [inline docs/specs/validation/metadata-validation.yml:530]:
```suggestion
    robots: noindex
```

File: src/docfx/validation/CustomRuleProvider.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/validation/CustomRuleProvider.cs:58]: The test cases is wrong :)

Comment (fkpwolf) [reply src/docfx/validation/CustomRuleProvider.cs]: good catch!😀

File: src/docfx/validation/CustomRuleProvider.cs (hunk 2 above)
Comment (fkpwolf) [reply src/docfx/validation/CustomRuleProvider.cs]: sure

File: docs/specs/validation/metadata-validation.yml (hunk 3 above)
Comment (fkpwolf) [reply docs/specs/validation/metadata-validation.yml]: ok

**Review by yufeih — APPROVED** [review]

**Comment by fkpwolf** [comment]
The diff shows now there are more `ms-author-invalid` suggestions. It should not be relevant to this PR. Some changes on Graph API? @yufeih

**Comment by yufeih** [comment]
> The diff shows now there are more `ms-author-invalid` suggestions. It should not be relevant to this PR. Some changes on Graph API? @yufeih

No. The new suggestions looks correct. Ship it!

**Pull request merged**